test/robot/web/build.go

```diff
@@ -54,6 +57,38 @@ func (s *Server) handlePackages(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (s *Server) handlePackageChain(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	head := r.FormValue("head")
+	if head == "" {
+		writeError(w, 404, errors.New("The head parameter is required"))
+		return
+	}
+
+	pkgs := make(map[string]*build.Package)
+	if err := s.Build.SearchPackages(ctx, q.Bool(true).Query(), func(ctx context.Context, entry *build.Package) error {
+		pkgs[entry.Id] = entry
+		return nil
+	}); err != nil {
+		writeError(w, 500, err)
+		return
+	}
+
+	pkg := pkgs[head]
```

Comment:
nit: Use `pkg, ok := pkgs[head]` and test `ok`, unless you expect `SearchPackages` to return `nil` for `entry`.